classes/class-uninstall.php

```diff
@@ -52,6 +52,8 @@ public function __construct( $plugin ) {
 	 * Uninstall Stream by deleting its data
 	 */
 	public function uninstall() {
+		check_ajax_referer( 'stream_uninstall_nonce', 'nonce' );
```

Comment:
This is the actual security fix. Most of the rest of the PR is about reliable testing to make sure the fix does what it is supposed to do.